src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcher.scala

```diff
@@ -188,42 +189,44 @@ class BlockFetcher(
         case Left(_) => state
         case Right(validHashes) => state.withPossibleNewTopAt(validHashes.lastOption.map(_.number))
       }
-
       supervisor ! ProgressProtocol.GotNewBlock(newState.knownTop)
-
       fetchBlocks(newState)
     case MessageFromPeer(NewBlock(_, block, _), peerId) =>
```

Comment:
can we rewrite this handling before take decisions about a refactor, i mean, every `if / if else` below uses different things, and is really hard to reason about it.. i would prefer to have something like:
```
// use a variable to reflect this condition instead of adding a comment
if (newBlockNr == nextExpectedBlock) {
  if(state.isOnTop) {
    ...
  } else {
     ...
  }
```
} else if (newBlockNr > nextExpectedBlock) {
  ...
} else {
 log.info("older block")
}